File: data/src/main/java/org/cryptomator/data/cloud/local/file/LocalStorageNodeFactory.java
package org.cryptomator.data.cloud.local.file;

import org.cryptomator.util.Optional;

import java.io.File;
import java.util.Date;

class LocalStorageNodeFactory {

	public static LocalNode from(LocalFolder parent, File file) {
		if (file.isDirectory()) {
			return folder(parent, file);
		} else {
			return file( //
					parent, //
					file.getName(), //
					file.getPath(), //
					Optional.of(file.length()), //
					Optional.of(new Date(file.lastModified())));
		}
	}

	public static LocalFolder folder(LocalFolder parent, File file) {
		return folder(parent, file.getName(), file.getPath());
	}

	public static LocalFolder folder(LocalFolder parent, String name, String path) {
		return new LocalFolder(parent, name, path);
	}

	public static LocalFile file(LocalFolder folder, String name, String path, Optional<Long> size, Optional<Date> modified) {
		return new LocalFile(folder, name, path, size, modified);
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/local/file/LocalStorageNodeFactory.kt
package org.cryptomator.data.cloud.local.file

import java.io.File
import java.util.Date

internal object LocalStorageNodeFactory {

	@JvmStatic
	fun from(parent: LocalFolder, file: File): LocalNode {
		return if (file.isDirectory) {
			folder(parent, file)
		} else {
			file( //
				parent,  //
				file.name,  //
				file.path,  //
				file.length(),  //
				Date(file.lastModified())
			)
		}
	}

	fun folder(parent: LocalFolder, file: File): LocalFolder {
		return folder(parent, file.name, file.path)
	}

	@JvmStatic
	fun folder(parent: LocalFolder, name: String, path: String): LocalFolder {
		return LocalFolder(parent, name, path)
	}

	@JvmStatic
	fun file(folder: LocalFolder, name: String, path: String, size: Long?, modified: Date?): LocalFile {
		return LocalFile(folder, name, path, size, modified)
	}
}
